news/2024/2024-09-17-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,176 @@
+
+---
+
+## osu!
+
+All changes for osu! are covered in the osu!(lazer) update video, which you may be interested in watching:
+
+<div align="center">
+    <iframe width="95%" style="aspect-ratio: 16 / 9;" src="https://www.youtube.com/embed/Kqu4TUOO5IY" frameborder="0" allowfullscreen></iframe>
+</div>
+
+As a reminder of the various core values in the difficulty calculation algorithm:
+
+- **Aim**: how difficult it is to consistently hit notes back-to-back (affected by approach rate, jumps, Hidden, Hard Rock and more)
+- **Speed**: how quickly new elements are presented to the player (affected by note density, BPM, Double Time and Half Time)
+- **Accuracy**: how difficult it is to stay rhythmically on time (affected by Hidden, Hard Rock, Flashlight and more)
+- **Strain**: the difficulty of enduring intense or technically demanding patterns often and for a long time (affected by streams, fast jumps and more)
+
+### Combo scaling removal
+
+The most monumental [change](https://github.com/ppy/osu/pull/16280) proposed by [moonpoint](https://osu.ppy.sh/users/9558549) – with improvements made by [KermitNuggies](https://osu.ppy.sh/users/33452559) – removes the "combo scaling factor" applied to the aim and speed portions of a score's PP.
+
+This change means a 1 miss score with a max combo of 500 compared to a 1 miss score with a max combo of 700 on the same beatmap will be punished equally. The miss penalty was also rewritten in order to accommodate these changes - misses have generally been made harsher, with the initial miss having a harsher penalty in order to differentiate between FCs and non-FCs.
+
+The miss penalty scaling is now based on the amount of difficult strains instead of object count. As the penalty is based upon the amount of difficult strains, this means longer maps with a lot of "filler" sections will be punished harder than longer beatmaps which are consistently difficult even if both beatmaps have similar object counts.
+
+There were concerns raised about consistency being less important with these changes as combo is no longer a factor, however this isn't an issue - using combo as a metric doesn't make much sense as the performance calculator does not know what sections of a beatmap that a score got misses on. As a result, we model any misses as being on the most difficult part.
+
+### Rhythm complexity improvements
+
+A [change](https://github.com/ppy/osu/pull/28871) proposed by [StanR](https://osu.ppy.sh/users/7217455) fixes some flaws in the rhythm complexity calculations.
+
+As a reminder of how rhythm works, notes are grouped into 'islands' (for example, a triple pattern is described as a 3 note island) and complexity is assessed using these islands.
+
+The first improvement in this change is removing a size bonus on island length and uncapping the island size. Previously, there was a bonus applied based on island length sizes that would cause long bursts to gain a lot of rhythm complexity even if they were not complex.
```

Comment:
this entire paragraph also seems very complex reading it as someone who's not a dev/familiar with the relevant formulas, also you introduce island "length" and "size" as two separate parameters then you mush them together the very next sentence:
`[...]island length and uncapping the island size.`
`[...]island length sizes[...]`

...that made me even more confused, so i would suggest maybe dumbing this entire portion down, rewording "islands" into "bursts" would make this a lot more comprehensible to the average reader (if applicable)

Reply:
bursts is not really fair, because it is grouping patterns together that might not necessarily be bursts. I think island is the right word (it's also the word the pp system uses). I'm not really sure what to do with this?

Reply:
i'd argue that how its referred to in the pp system isnt relevant to the end-user, for me personally when i just swapped the word islands with bursts the whole paragraph made a lot more sense, but i understand that *technically* it isnt the correct term

an argument could be made that despite the actual term being used it might be better to dumb it down to something more easily understandable for the general audience, especially since the term "island" is completely foreign to the general public

Reply:
i'm fine dumbing it down, i just think using a word like "burst" might cause more confusion because it's a grouping of patterns - could even be a jump for example

Reply:
that's ok, you dont have to use that exact wording, let me know when you draft something up in the discord and ill take a look

Reply:
tbf the original newspost explaining the rhythm system used `islands` as well https://github.com/ppy/osu-wiki/blob/master/news/2021/2021-11-09-performance-points-star-rating-updates.md#rhythm-complexity

Reply:
in that case using that word is fine, and after sleeping over it maybe the same explanation as that post can be used (the three bullet points), then go into what's been changed

Reply:
I've re-used the explanation from that previous post about islands instead of what I had before